### PR TITLE
fix(List/ColumnChooser): "Select all" and columns list filter

### PR DESCRIPTION
**Pull request by sgangnant**
**What is the problem this PR is trying to solve?**
"Select all" option in the lists' column choser should apply to shown columns (e.g. those shown when the user filters the list of columns)
image

**What is the chosen solution to this problem?**
Integrate filtering into the column chooser manager and update the manager's behavior accordingly, example :
- Given a list with 3 columns, "_First name_", "_Last name_", "_Login_", all 3 checked (so "_Select all_" is also checked)
- User types in "_name_" in the filter (only "_First name_" and "_Last name_" appear in the list) **→** Unchecks "_Select all_" **→** Apply
- Only "_Login_" column is shown in the list

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/List/Toolbar/ColumnChooserButton/hooks/columnChooserManager.hook.js [hunk 1]
```diff
@@ -1,26 +1,7 @@
...
- * Helps to change the select all visible status. If all columns are visible, select all is checked.
- * @param {array} columns
- */
-const isEveryItemVisible = columns => columns.every(isItemVisible);
-
-const findColumnLabel = label => column => column.label === label;
-
-export const getColumn = label => columns => columns.find(findColumnLabel(label));
-
-export const setColumn = (column, label) => columns => {
```
Comment (jmfrancois) [inline packages/components/src/List/Toolbar/ColumnChooserButton/hooks/columnChooserManager.hook.js:18]: it looks like a breaking change but they do not seems to be used outside. could you please confirm ?

**Review by jmfrancois — APPROVED** [review]
The story is working as described and the tests looks good !

File: packages/components/src/List/Toolbar/ColumnChooserButton/hooks/columnChooserManager.hook.js (hunk 1 above)
Comment (sgangnant) [reply packages/components/src/List/Toolbar/ColumnChooserButton/hooks/columnChooserManager.hook.js:18]: I couldn't find a place where it was used outside the components so I assume it's safe to remove it (don't know why it was exported prior this PR, tests ? 🤔)

File: packages/components/src/List/Toolbar/ColumnChooserButton/hooks/columnChooserManager.hook.js [hunk 2]
```diff
@@ -89,33 +71,56 @@ const updateVisibilityAttr = changeColumnChooserAttribute('visible');
...
+export const useColumnChooserManager = (
+	initialColumns = [],
+	nbLockedLeftItems = 0,
+	initialFilterValue = '',
+) => {
+	const [columns, setColumns] = useState(() => prepareColumns(initialColumns, nbLockedLeftItems));
+
+	const [textFilter, setTextFilter] = useState(initialFilterValue);
+
+	const filteredColumns = useMemo(
```
Comment (frassinier) [inline packages/components/src/List/Toolbar/ColumnChooserButton/hooks/columnChooserManager.hook.js:83]: Ok not the best way to filter but it was just moved

**Review by frassinier — APPROVED** [review]

**Pull request merged**